Pachinko Description (#24)
MrLongbottom/P10-Article: sections/plate_notation.tex

	\label{fig:metadata_lda}
\end{figure*}

\subsection{Pachinko Allocation}

> In order to handle the hierarchical structure of the taxonomy metadata field, we will be using a hierarchical topic model, namely pachinko \citet{li2006pachinko}.

Reviewer: ```suggestion
In order to handle the hierarchical structure of the taxonomy metadata field, we will be using a hierarchical topic model, namely the \acrlong{pam} from \citet{li2006pachinko}.
```

> Pachinko allocation generalizes \gls{lda}, to be able to operate on any DAG structure.

Reviewer: ```suggestion
Pachinko allocation generalizes \gls{lda}, to be able to operate on any \gls{dag} structure.
```

> The DAG structure represents topics which in pachinko are distributions over words and/or other topics.

Reviewer: ```suggestion
The \gls{dag} structure represents topics which in pachinko are distributions over words and/or other topics.
```

It is set up in a tree structure with the first layer being the root node, a topic which all documents are part of, and the bottom layer consisting of leaf nodes, which are words. 

For simplicity, we will be using a five-level pachinko tree structure, with each node in a layer having an edge to each node in the next layer.

> We therefore also limit the taxonomy chain to the first $3$ taxonomy entries, to fit with our DAG structure.

Reviewer: ```suggestion
We therefore also limit the taxonomy chain to the first $3$ taxonomy entries, to fit with our \gls{dag} structure.
```

> This DAG-structure is visualizated in \autoref{fig:pachinko_dag}

Reviewer: ```suggestion
This \gls{dag} structure is visualized in \autoref{fig:pachinko_dag}.
```

For each word, a chain of topics is sampled by sampling one topic from each layer of the structure, and finally, a word is chosen based on the last topic.

> More specifically, for each topic starting with the root topic, we sample a topic from the next layer $Z_{wl}$ from that node's topic distribution $\theta_l^(d)$ based on observed data from that document.

Reviewer: ```suggestion
More specifically, for each topic starting with the root topic, we sample a topic from the next layer $Z_{wl}$ from that node's topic distribution $\theta_l^{(d)}$ based on observed data from that document.
```

Reviewer: ```suggestion
More specifically, for each topic starting with the root topic, we sample a topic from the next layer $Z_{wl+1}$ from that node's topic distribution $\theta_l^{(d)}$ based on observed data from that document.
```
Man kunne også måske tilføje '+1' ved l for at tydeliggøre at det er det næste lag.

> 

Reviewer: ```suggestion
The plate notation for the \gls{pam} can be seen in \autoref{fig:pachinko}.

```
Da vi ellers ikke nævner figuren i teksten.

\begin{figure}
	\centering
	\includegraphics[width= 0.5 \linewidth]{figures/pachinko_dag.png}

> 	\caption{Illustration of the DAG-structure for five-level PAM}

Reviewer: ```suggestion
	\caption{Illustration of the \gls{dag} structure for the five-level \gls{pam}.}
```

	\label{fig:pachinko_dag}
\end{figure}

\input{figures/pachinko}
\todo[inline]{make illustrations of dag structur pretty}

\subsubsection{Modification to Pachinko Allocation}

> Since only $25\%$ of the documents have an observed taxonomy, we can then lock the words in the observed documents to be in the corresponding topics within the pachinko DAG instead of continuously sampling them using Gibbs sampling.

Reviewer: ```suggestion
Since only $25\%$ of the documents have an observed taxonomy, we can then lock the words in the observed documents to be in the corresponding topics within the pachinko \gls{dag} instead of continuously sampling them using Gibbs sampling.
```

> This creates some constant context for the topics, which the documents with unobserved taxonomy will be fitted around.

Reviewer: ```suggestion
This creates some constant context for the topics, which the documents with unobserved taxonomies will be fitted around.
```

> Some of the observed documents have multiple taxonomies. For these documents, one of the taxonomies is chosen randomly for each word in the document.

Reviewer: ```suggestion
Some of the observed documents have multiple taxonomies.
For these documents, one of the taxonomies is chosen randomly for each word in the document.
```

	\label{fig:metadata_lda}
\end{figure*}

\subsection{Pachinko Allocation}

> In order to handle the hierarchical structure of the taxonomy metadata field, we will be using a hierarchical topic model, namely pachinko \citet{li2006pachinko}.

Reviewer 2: ```suggestion
To handle the hierarchical structure of the taxonomy metadata field, we will be using a hierarchical topic model, namely pachinko \citet{li2006pachinko}.
```

> Pachinko allocation generalizes \gls{lda}, to be able to operate on any DAG-structure.

Reviewer 2: ```suggestion
Pachinko allocation generalizes \gls{lda}, to be able to operate on any DAG structure.
```
måske lav DAG til en gls?

The DAG structure represents topics which in pachinko are distributions over words and/or other topics.

> It setup in a tree structure with the first layer being the root node, a topic which all documents are part of, and the bottom layer consisting of leaf nodes which are words. 

Reviewer 2: ```suggestion
It is set up in a tree structure with the first layer being the root node, a topic which all documents are part of, and the bottom layer consisting of leaf nodes, which are words. 
```

> For simplicity we will be using five-level pachinko tree structure, with each node in a layer having an edge to each node in the next layer.

Reviewer 2: ```suggestion
For simplicity, we will be using a five-level pachinko tree structure, with each node in a layer having an edge to each node in the next layer.
```

> We also therefore limit the taxonomy chain to the first $3$ taxonomy entries, in order to fit with our DAG-structure.

Reviewer 2: ```suggestion
We therefore also limit the taxonomy chain to the first $3$ taxonomy entries, to fit with our DAG structure.
```

This DAG-structure is visualizated in \autoref{fig:pachinko_dag}

> For each word a chain of topics is sample by sampling one topic from each layer of the structure, and finally a word is chosen based on the last topic.

Reviewer 2: ```suggestion
For each word, a chain of topics is sampled by sampling one topic from each layer of the structure, and finally, a word is chosen based on the last topic.
```

> More specifically, for each topic starting with the root topic, we sample a topic from the next layer $Z_{wl}$ from that node's topic distribution $\theta_l^(d)$ based on observed data from that document.

Reviewer 2: ```suggestion
More specifically, for each topic starting with the root topic, we sample a topic from the next layer $Z_{wl}$ from that node's topic distribution $\theta_l^(d)$ based on observed data from that document.
```

Author: no change?

> 

Reviewer 2: Det kunne være lækkert med et simpelt eksempel, like sample Steder -> Denmark -> Aalborg ?

Author: ```suggestion
\todo[inline]{add example (steder -> denmark -> aalborg)}
```

\begin{figure}
	\centering
	\includegraphics[width= 0.5 \linewidth]{figures/pachinko_dag.png}
	\caption{Illustration of the DAG-structure for five-level PAM}
	\label{fig:pachinko_dag}
\end{figure}

\begin{figure}
	\centering
	\includegraphics[width= 0.9 \linewidth]{figures/five_layer_pachinko.png}
	\caption{Plate notation for five-level PAM}
	\label{fig:pachinko_plate}
\end{figure}
\todo[inline]{make pretty picture for pachinko plate notation}

\subsubsection{Modification to Pachinko Allocation}

> Since only $25\%$ of the documents have an observed taxonomy, we can then lock the words in the observed documents to be in the corresponding topics within the pachinko DAG instead of conteniously sampling them using gibbs sampling.

Reviewer 2: ```suggestion
Since only $25\%$ of the documents have an observed taxonomy, we can then lock the words in the observed documents to be in the corresponding topics within the pachinko DAG instead of continuously sampling them using Gibbs sampling.
```

\end{figure*}

\subsection{Pachinko Allocation}
In order to handle the hierarchical structure of the taxonomy metadata field, we will be using a hierarchical topic model, namely the \acrlong{pam} from \citet{li2006pachinko}.
Pachinko allocation generalizes \gls{lda}, to be able to operate on any \gls{dag} structure.

> The \gls{dag} structure represents topics which in pachinko allocation are distributions over words and/or other topics further down in the \gls{dag} structure, rather than distribution only over words as with \gls{lda}.

Reviewer: ```suggestion
The \gls{dag} structure represents topics which in pachinko allocation are distributions over words and/or other topics further down in the \gls{dag} structure, rather than a distribution only over words as with \gls{lda}.
```

It is set up in a tree structure with the first layer being the root node, a topic which all documents are part of, and the bottom layer consisting of leaf nodes, which are words.
The middle layers will represent the taxonomy entries in our dataset.

For simplicity, we will be using a five-level pachinko tree structure, with each node in a layer having an edge to each node in the next layer.

> We therefore also limit the taxonomy dataset to the only first $3$ taxonomy entries for each document, to fit with our \gls{dag} structure.

Reviewer: ```suggestion
We therefore also limit the taxonomy dataset to only the first $3$ taxonomy entries for each document, to fit with our \gls{dag} structure.
```

This \gls{dag} structure is visualized in \autoref{fig:pachinko_dag}.

For each word, a chain of topics is sampled by sampling one topic from each layer of the structure, and finally, a word is chosen based on the last topic.

> More specifically, for each topic starting with the root topic, we sample a topic from the next layer $Z_{wl+1}$ from that node's topic distribution $\theta_l^{(d)}$ based on observed data from that document.

Reviewer: ```suggestion
More specifically, for each topic starting with the root topic, we sample a topic from the next layer $Z_{w(l+1)}$ from that node's topic distribution $\theta_l^{(d)}$ based on observed data from that document.
```
